### PR TITLE
Build and test through Github Actions

### PR DESCRIPTION
**Pull request by francoislehoux-okta**
## Description:
- Replaces travis build pipeline with a github action

### Resolves:
* [OKTA-557843](https://oktainc.atlassian.net/browse/OKTA-557843)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,31 +0,0 @@
...
-  - SAUCE_BROWSER_NAME="chrome"
-  - SAUCE_BROWSER_VERSION=""
-  - SAUCE_PLATFORM="Windows 10"
-  - SAUCE_APPIUM_VERSION=""
-  - SAUCE_DEVICE_NAME=""
-  - SAUCE_DEVICE_ORIENTATION=""
-  - SAUCE_PLATFORM_VERSION=""
-  - SAUCE_PLATFORM_NAME=""
-  - secure: RW99TW7YvGgWt/P8fQQziPbxNITaJMO/4q2iCX72u/PtTu2dkeaCqUiRUi7qyV8jIIk7U4oruxJRFQpTaCIX5+JWZsSwLskoTsspOI9aTzyPdhVNckM1vpA5VfwstdyQTkX7zzKPLdvXK0xj0ZzQ+ACH/SobQNYKiAMvW7FVkIc=
-  - secure: gFSORYQLSBNIK8peGgUOhd0Yj6GzE18HA0GLrmp0PMtobe003poalkNgLsPy2W2QFZbG3bCjbqFMmCbUh5uRnX8eCy/whe+lwXeQsv5uh0EIGWgxVxhk79dCx5cakERusoj2BUgJJ432R7nRXjEW5v2AK9QAOeCG2Dvwdb8WSes=
```
Comment (francoislehoux-okta) [inline .travis.yml:25]: Could not find usage of these anywhere. Asked around and no one knows if this was being used or not.

File: .travis.yml [hunk 2]
```diff
@@ -1,31 +0,0 @@
...
-notifications:
-  slack:
-    secure: FbvUST5p63eiqJa4dxArVFYzTittyKzSt0+StBEaJwvLtn/ZiW4N2bJovtccThyIMGny0NJz34rh1jlVXUXOzWYWVTgr/fK4vWhN8mJWK17gg5aIfJr9oJ029k/cZB6luJzY6J8Ld/5SZ7Xsi2SE6Dbnntdx9e6gwSs3f8YqKF8qsPeb4Jw07ax6AR+MgSc3vyjZZeR8g6FjxLTzMepSsbfGs7vrmJO9znW+lhf0mYJaHwlaVLt9mMlgfeciCa9HImqVHID597XvcKZpeXq3OL6NCwE1dQcPjB5/xR4Gl3K9I2bOn9gHf6tZf9di0HD8lMnQ71pc45JYOzs3UTQwdQNNjEyG/ZF0YT7Mok0dcMSDUDCXYWpgTohrz5LzXNKyOYAS3FXIfxQwfegFC065h4/Ex1pSVQCB+7/v/qBBkBZuyLhUfATDvVq22zduK3Q+X4dEwzxvNfCwpOvYe4xPdFUVcNUVFCa9D5VZnbUdYZiF5k1daVL7iRTTT17fc4ssfposFdkh+IX332Cb6y6ytkRBchTW64mk0qQ0haLMfDygn2uzfX0orr2M+S06pQ8oIlYkaffoi2UB1C2jHWFxB73/tbT6ivW0NWj22m/7U7y2Enx/qOOy3LL/SQNhaVOy+UIzWrY7wm+2AlNbs1O5pZ88VJQZZKzDkXx7jb/R7hs=
-  on_pull_requests: false
-  on_success: never
-  on_failure: always
```
Comment (francoislehoux-okta) [inline .travis.yml:31]: I have never seen notifications from travis on slack for dev.okta

File: .travis.yml [hunk 3]
```diff
@@ -1,31 +0,0 @@
...
-node_js:
-- '14'
-addons:
-  chrome: stable
-before_install: "source scripts/helpers.sh"
-install: yarn install
-before_script: "yarn build"
-script:
-  - "./scripts/travis.sh"
-dist: xenial
```
Comment (francoislehoux-okta) [inline .travis.yml:12]: Using `ubuntu-latest` seems to work without having to define a particular dist. Not even sure we could do it with github actions

File: scripts/helpers.sh [hunk 4]
```diff
@@ -112,13 +112,13 @@ function check_for_quickstart_pages_in_sitemap() {
...
-function fold() {
+function group() {
     local name=$1
     local command="${@:2}"
-    echo -en "travis_fold:start:${name}\\r"
+    echo -en "::group::${name}\\r"
     echo "\$ ${command}"
     ${command}
-    echo -en "travis_fold:end:${name}\\r"
+    echo -en "::endgroup::${name}\\r"
```
Comment (francoislehoux-okta) [inline scripts/helpers.sh:121]: Confirmation that this works:

Screen Shot 2022-12-09 at 6 00 42 PM

File: .travis.yml (hunk 2 above)
Comment (brianduffield-okta) [reply .travis.yml:31]: We do receive broken Travis alerts via #eng-devex-alerts ...not sure this is it.

Comment (francoislehoux-okta) [reply .travis.yml:31]: Will add a follow up to figure out sending failures through slack

**Review by shawyu-okta — APPROVED** [review]

File: scripts/yarn-checks.sh [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+# Will run the netlify deploy if:
+# - NETLIFY_SITE_ID and NETLIFY_AUTH_TOKEN is present (will not be true on forks)
+# - The latest commit message on the PR includes the text NFDEPLOY
+# if [ -n $NETLIFY_SITE_ID ] && git log -n1 --pretty=format:"%B" $TRAVIS_PULL_REQUEST_SHA | grep -q NFDEPLOY; then
+#     group netlify_setup npm install -g netlify-cli@^12
+#     group netlify netlify deploy
+# else
+#     echo "Will not use Netlify deploy"
+# fi
```
Comment (francoislehoux-okta) [inline scripts/yarn-checks.sh:24]: https://oktainc.atlassian.net/browse/OKTA-558239

**Review by robertdamphousse-okta — APPROVED** [review]

**Pull request merged**